### PR TITLE
Add new options includeFiles

### PR DESCRIPTION
**Pull request by MaximusFT**
I hope all my changes will be clear.
Add tests for all new cases. All test is green
image

Feel free to write in any question.

- minor refactor
- add option.includeFiles
- add option.bethSymbol
- add tests to cover new features
- update Readme

PS: for me important to use an updated version in my project )

### CODE REVIEW

**Comment by MaximusFT** [comment]
These changes can close #77 issue.

**Comment by RadValentin** [comment]
I'm not sure I understand how `bethSymbol` is supposed to be used. To get this result `.hello+.a`, which is the same as `.hello + .a` you could just set the existing prefix param to: `prefix: '.hello +'`. AFAIK CSS doesn't care if there are spaces or not.

For gluing the prefix to the selector (`.hello.a`), the transform param can be used:

```
transform(prefix, selector, prefixedSelector) {
  return prefix + selector;
}
```

Since both the cases you presented are already covered by existing functionality then I'd avoid adding more params to the plugin.

File: index.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
-  const prefix = options.prefix;
-  const prefixWithSpace = /\s+$/.test(prefix) ? prefix : `${prefix} `;
+  const { prefix, bethSymbol = ' ' } = options;
+  const prefixWithSymbol = prefix.trim() + bethSymbol;
   const ignoreFiles = options.ignoreFiles ? [].concat(options.ignoreFiles) : [];
+  const includeFiles = options.includeFiles
+    ? [].concat(options.includeFiles)
+    : [];
+
+  const searchSelector = (selector, searchArray) => {
```
Comment (RadValentin) [inline index.js:9]: I'd like to keep this as a separate function

**Review by RadValentin — CHANGES_REQUESTED** [review]
I'd remove `bethSelector` since it's covered by existing functionality.

**Comment by MaximusFT** [comment]
> > These changes can help to close #77 issue.
> 
> That issue was referring to prefixing only certain selectors, not certain files so sadly it will remain open :(

Yes, you right... I didn't pay enough attention

**Comment by MaximusFT** [comment]
Update PR.
Leave new tests and solve them as you propose.

File: index.js [hunk 2]
```diff
@@ -10,6 +13,12 @@ module.exports = function postcssPrefixSelector(options) {
     ) {
       return;
     }
+    if (
+      root.source.input.file &&
+      includeFiles.some((file) => !root.source.input.file.includes(file))
```
Comment (RadValentin) [inline index.js:18]: This won't work as expected, if the `includeFiles` array has two entries for example 'a.css' and `b.css` this will return `false`.

File: test/test.js [hunk 3]
```diff
@@ -115,6 +115,87 @@ it('should prefix selectors in unignored files', () => {
   assert.equal(out, expected);
 });
 
+it('should prefix selectors in included file', () => {
```
Comment (RadValentin) [inline test/test.js:118]: Please add tests for empty include array and include array with multiple files

File: test/test.js [hunk 4]
```diff
@@ -115,6 +115,87 @@ it('should prefix selectors in unignored files', () => {
...
+      })
+    )
+    .process(getFixtureContents('between-symbol-selector.css')).css;
+
+  const expected = getFixtureContents('between-symbol-selector.expected.css');
+
+  assert.equal(out, expected);
+});
+
+it('should use custom symbol between prefix and selector. Use "+"', () => {
```
Comment (RadValentin) [inline test/test.js:169]: Please remove this test, I don't see the value of having it anymore.

**Review by RadValentin — CHANGES_REQUESTED** [review]
PR looks better but you need to solve the issue I mentioned with how includeFiles works and update the tests.

**Comment by MaximusFT** [comment]
@RadValentin 
Thanks for the review.
Update PR

**Comment by RadValentin** [comment]
Looks good to me! 👍  Thanks for contributing, I'll release a new version of the plugin today.
